ros/angel_utils/python/angel_utils/activity_classification.py

```diff
@@ -192,13 +192,17 @@ def queue_image(
         # self.get_logger_fn().info(f"self.frames[-1][0] header stamp: {self.frames[-1][0]}")
         with self.__state_lock:
             # before the current lead frame?
-            if self.frames and time_to_int(img_header_stamp) <= time_to_int(
-                self.frames[-1][0]
+            if (
+                self.frames
+                and self.frames[-1][2] == image_frame_number
+                and time_to_int(img_header_stamp) <= time_to_int(self.frames[-1][0])
```

Comment:
Were we getting situations where different frames had the same timestamp?

Reply:
Yes, although I'm not sure if this is only an artifact of playing a bag that was created with our script.

Reply:
Looks like this was caused by our script that converted videos to bags - I've got a fix in for that now, so I've backed this change out, @Purg